src/LNLib/Geometry/Curve/NurbsCurve.cpp

```diff
@@ -3578,27 +3605,26 @@ std::vector<double> LNLib::NurbsCurve::GetParamsOnCurve(const LN_NurbsCurve& cur
 std::vector<LNLib::XYZ> LNLib::NurbsCurve::Tessellate(const LN_NurbsCurve& curve)
 {
 	int degree = curve.Degree;
-	std::vector<double> knotVector = curve.KnotVector;
-	std::vector<XYZW> controlPoints = curve.ControlPoints;
+	const std::vector<double>& knotVector = curve.KnotVector;
+	const std::vector<XYZW>& controlPoints = curve.ControlPoints;
 
 	if (degree == 1)
 	{
 		return ControlPointsUtils::ToXYZ(controlPoints);
 	}
 
-	double start = knotVector[0];
-	double end = knotVector[knotVector.size() - 1];
-
-	XYZ startPoint = controlPoints[0].ToXYZ(true);
-	XYZ endPoint = controlPoints[controlPoints.size() - 1].ToXYZ(true);
-
+	std::vector<double> unqiueKnotVector = knotVector;
+	unqiueKnotVector.erase(std::unique(unqiueKnotVector.begin(), unqiueKnotVector.end()), unqiueKnotVector.end());
+	
 	std::vector<double> parameters;
```

Comment:
Use std::set to facilitate sorting of parameters. Or reorganize the implementation of “TessellateCore” so that the parameters in the std::vector are arranged in ascending order.

Reply:
ok，I will use std::set

Reply:
Input curve's knotvector is nondecreasing sequence ( we suggest use Check() to check curve parameters ), thus here use unique.